[ES|QL] Fix lookup index autocomplete

sddonne opened the pull request:
Closes https://github.com/elastic/kibana/issues/235167
## Summary
Calculate correctly the range to replace for the lookup index editor suggestions.
We need to calculate them by hand since monaco splits words by dot or slash.

* Moved `getLookupIndexCreateSuggestion` from the function utils as it's not a function util.


https://github.com/user-attachments/assets/44ee8294-a881-43e3-88ba-8ca11614d336

sddonne committed:
fix funciton completions

sddonne committed:
calculate the corrrect rangeToReplace for lookup join suggestion

sddonne committed:
fix test

elasticmachine commented:
Pinging @elastic/kibana-esql (Team:ESQL)

sddonne committed:
Merge remote-tracking branch 'upstream/main' into fix-lookup-index-autocomplete

sddonne committed:
test special chars in index name

akowalska622 reviewed (approved):
Data Discovery changes (again) LGTM 😁  I thought it's on purpose on last PR, but reverting back looks even better to me

akowalska622 commented:
Perfect then! :)

drewdaemon commented:
The pictured bug is fixed. I do see a related one:

https://github.com/user-attachments/assets/11628414-5f97-4bf0-9cc3-8b1b65117313

It's okay if we want to handle that one elsewhere, though. Up to you!

sddonne committed:
handle indexes fragments

sddonne commented:
> The pictured bug is fixed. I do see a couple of related ones:
> 
>  Screen.Recording.2025-09-16.at.10.53.36.AM.mov 
>  suggests_new_index.mov 
> I'm okay if we want to handle those elsewhere, though they do seem related.

@drewdaemon @stratoula 

Fixed! Added also support for fragments of index names. 176a3f7e6359009e28e630e094355e86a70cd196

https://github.com/user-attachments/assets/1beb42e2-bf40-4255-9e45-3c5e20d33649

sddonne commented:
> Seems like the ranges are working for the existing index suggestions, but I still don't really understand when/why the "create new" suggestion pops up in some cases and not others
> 
>  Screen.Recording.2025-09-17.at.1.38.22.PM.mov 
>  Screen.Recording.2025-09-17.at.1.37.17.PM.mov 
> A wild guess: it could be that the text of the "create new" suggestion is being created but is not updated with later characters, leading it to get filtered out of the suggestion list. Not sure though.

@drewdaemon oh did not understood that was the issue you were pointing.
Yes.. it seems that monaco does not ask for new suggestions while you are typing within a word (if there is still a matching suggestion from the list it already has), so the `Create lookup index "{indexName}"` suggestion will only be updated with the `indexName` when you ask for suggestions manually hitting ctrl + space if you are typing an index with similar name.

On the contrary if you don't have any index created that starts with what you are typing, it updates the suggestions key stroke by key stroke.

Have been checking a little bit, to trigger completions on every keystroke we can use `onDidChangeModelContent` and trigger `editor.action.triggerSuggest` , but I think this is to much of a performance degradation just for this scenario.

Do you have other less aggressive solutions on top of your mind?

sddonne committed:
Merge branch 'main' into fix-lookup-index-autocomplete

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340885)
* Commit: 2c7bf00a74b421f014cfded470387d72b2d8cf59

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/340885#01995d3e-803c-493c-98b1-d6d02aadece7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340885#01995d3e-803c-493c-98b1-d6d02aadece7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340885/jobs/01995d3e-803c-493c-98b1-d6d02aadece7/artifacts/01995d55-2850-4c2c-8bd6-c24153ff9ed5) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - editing processors - should edit an existing processor

### History
* :green_heart: [Build #340240](https://buildkite.com/elastic/kibana-pull-request/builds/340240) succeeded 176a3f7e6359009e28e630e094355e86a70cd196
* :yellow_heart: [Build #339812](https://buildkite.com/elastic/kibana-pull-request/builds/339812) was flaky f07d3bd9dc0eec4ab8607833372cb0ef8e3acf70
* :green_heart: [Build #339703](https://buildkite.com/elastic/kibana-pull-request/builds/339703) succeeded c9bca02a355a292fca0a05bc86d88cd3ece41a7b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340885","number":340885,"commit":"2c7bf00a74b421f014cfded470387d72b2d8cf59"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340240","number":340240,"commit":"176a3f7e6359009e28e630e094355e86a70cd196"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339812","number":339812,"commit":"f07d3bd9dc0eec4ab8607833372cb0ef8e3acf70"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339703","number":339703,"commit":"c9bca02a355a292fca0a05bc86d88cd3ece41a7b"}],"number":340885}
buildkite-pr-comment-->

sddonne committed:
add incomplete flag to suggestions

drewdaemon reviewed (approved):
Works great, @sddonne !

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340932)
* Commit: c68858896ff0eab466d131b7e8bae5858183724a

### Failed CI Steps
* [FTR Configs #118](https://buildkite.com/elastic/kibana-pull-request/builds/340932#01995d98-5da8-4765-8656-225ebee44189)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/64e7fa0c6bfd3ce3626e4dcf273c51149fe220da) | [after](https://github.com/elastic/kibana/commit/c68858896ff0eab466d131b7e8bae5858183724a) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.9MB | 3.9MB | +313.0B |



### History
* :broken_heart: [Build #340885](https://buildkite.com/elastic/kibana-pull-request/builds/340885) failed 2c7bf00a74b421f014cfded470387d72b2d8cf59
* :green_heart: [Build #340240](https://buildkite.com/elastic/kibana-pull-request/builds/340240) succeeded 176a3f7e6359009e28e630e094355e86a70cd196
* :yellow_heart: [Build #339812](https://buildkite.com/elastic/kibana-pull-request/builds/339812) was flaky f07d3bd9dc0eec4ab8607833372cb0ef8e3acf70
* :green_heart: [Build #339703](https://buildkite.com/elastic/kibana-pull-request/builds/339703) succeeded c9bca02a355a292fca0a05bc86d88cd3ece41a7b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340932","number":340932,"commit":"c68858896ff0eab466d131b7e8bae5858183724a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340885","number":340885,"commit":"2c7bf00a74b421f014cfded470387d72b2d8cf59"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340240","number":340240,"commit":"176a3f7e6359009e28e630e094355e86a70cd196"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339812","number":339812,"commit":"f07d3bd9dc0eec4ab8607833372cb0ef8e3acf70"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339703","number":339703,"commit":"c9bca02a355a292fca0a05bc86d88cd3ece41a7b"}],"number":340932}
buildkite-pr-comment-->

paulinashakirova reviewed (approved):
code review only
lgtm
🚀

sddonne committed:
update with main
